src/components/Map.vue

```diff
@@ -276,15 +275,39 @@ export default {
         });
       },
     },
+    isMrXVisible: {
+      handler() {
+        const svg = document.getElementById("board");
+        if (this.isMrXVisible) {
+          // check if user has access to Mr X
+          const isMrXTurn =
+            this.currentPlayer && this.currentPlayer.role === "mr-x";
+
+          if (!isMrXTurn && this.isMrXVisible && this.mrXLocation) {
+            const mrXLocationStation = this.$refs[this.mrXLocation][0];
```

Comment:
Bro can you explain this line in detail?

Reply:
same as above, I'm basically getting the exact circle DOM element (which is stored as a `ref`) see comment above:

https://github.com/azharkhan/scotland-yard/blob/master/src/components/Map.vue#L17